### PR TITLE
MM-53355: tidy up plugins

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Some documentation and organization prior to changing how deprecated, pre-packaged plugins work:
* remove support for feature flag managed plugins (we don't use this anymore)
* remove the plugin blocklist (product mode isn't a thing)
* avoid the use of the global logger
* leverage wrapped loggers (e.g. more consistently logging `plugin_id`)
* promote some logs from `Debug` to `Info` for better visibility

Note that this focuses on just `plugin.go` and `plugin_install.go`, and tries to avoid moving code (we can reorganize the functions into files at at a later date).

There should be no user-facing changes as a result of this PR.

#### Ticket Link
Relates-to: https://mattermost.atlassian.net/browse/MM-53355

#### Release Note
```release-note
Improved logging on plugin initialization, activation, and removal.
```

### CODE REVIEW

File: server/channels/app/plugin.go [hunk 1]
```diff
@@ -30,11 +30,14 @@ import (
...
 
+// prepackagedPluginsDir is the hard-coded folder name where prepackaged plugins are bundled
+// alongside the server.
 const prepackagedPluginsDir = "prepackaged_plugins"
 
+// pluginSignaturePath tracks the path to the plugin bundle and signature for the given plugin.
 type pluginSignaturePath struct {
 	pluginID      string
-	path          string
+	bundlePath    string
```
Comment (lieut-data) [inline server/channels/app/plugin.go:40]: Converging on the term `bundle` to represent the `*.tar.gz`.

File: server/channels/app/plugin_install.go [hunk 2]
```diff
@@ -59,88 +58,93 @@ const managedPluginFileName = ".filestore"
...
+// servers to do the same.
+func (ch *Channels) removePluginFromClusterMessage(pluginID string) {
+	logger := ch.srv.Log().With(mlog.String("plugin_id", pluginID))
+
+	logger.Info("Removing plugin as per cluster message")
 
-	if err := ch.removePluginLocally(data.Id); err != nil {
-		mlog.Warn("Failed to remove plugin locally", mlog.Err(err), mlog.String("id", data.Id))
+	if err := ch.removePluginLocally(pluginID); err != nil {
+		logger.Error("Failed to remove plugin locally", mlog.Err(err))
```
Comment (lieut-data) [inline server/channels/app/plugin_install.go:122]: Something may be out of sync here if this happens, so promoting to an error.

File: server/channels/app/plugin_install.go [hunk 3]
```diff
@@ -149,17 +153,42 @@ func (ch *Channels) installPlugin(pluginFile, signature io.ReadSeeker, installat
...
+	if err := ch.notifyPluginStatusesChanged(); err != nil {
+		logger.Warn("Failed to notify plugin status changed", mlog.Err(err))
+	}
+
+	return manifest, nil
+}
+
+// installPluginToFilestore saves the given plugin bundle (optionally signed) to the filestore,
+// notifying cluster peers accordingly.
+func (ch *Channels) installPluginToFilestore(manifest *model.Manifest, bundle, signature io.ReadSeeker) *model.AppError {
```
Comment (lieut-data) [inline server/channels/app/plugin_install.go:176]: Just a refactor of the above method -- the same sequence is called.

File: server/channels/app/plugin_install.go [hunk 4]
```diff
@@ -245,7 +269,7 @@ func (ch *Channels) InstallMarketplacePlugin(request *model.InstallMarketplacePl
 		return nil, model.NewAppError("InstallMarketplacePlugin", "app.plugin.marketplace_plugins.signature_not_found.app_error", nil, "", http.StatusInternalServerError)
 	}
 
-	manifest, appErr := ch.installPluginWithSignature(pluginFile, signatureFile)
+	manifest, appErr := ch.installPlugin(pluginFile, signatureFile, installPluginLocallyAlways)
```
Comment (lieut-data) [inline server/channels/app/plugin_install.go:272]: Avoiding a wrapper here and just passing the installation strategy directly.

**Comment by lieut-data** [comment]
@crspeller & @mickmister, you may find the individual commits easier to review. Your call!

**Review by crspeller — APPROVED** [review]
Nice cleanup.

**Review by mickmister — APPROVED** [review]
Excellent work @lieut-data :tada:

**Comment by lieut-data** [comment]
@amyblais, with no related e2e tests failing, I'm going to defer QA review on this delta and submit the PR with the prepackaged plugin rework that will require QA review.

**Pull request merged**